activesupport/lib/active_support/core_ext/range/each.rb

```diff
@@ -11,13 +11,45 @@ def each(&block)
 
     def step(n = 1, &block)
       ensure_iteration_allowed
-      super
+
+      case n
+      when ActiveSupport::Duration
+        duration_step(n, &block)
+      else
+        super
+      end
     end
 
     private
       def ensure_iteration_allowed
+        raise TypeError, "can't iterate beginless range" unless self.begin
         raise TypeError, "can't iterate from #{first.class}" if first.is_a?(TimeWithZone)
```

Comment:
Currently, beginless ranges fail with this error:
```ruby
(..5).step(2)
# cannot get the first element of beginless range
```

This is because [calling `Range#first` on a beginless range raises an error.](https://docs.ruby-lang.org/en/master/Range.html#method-i-first)

The additional error aims to make the error message more useful.

Note also Ruby behavior without ActiveSupport:
```ruby
(..5).step(2).to_a
# nil can't be coerced into Integer (TypeError)
```